cpp/include/cudf/utilities/span.hpp

```diff
@@ -43,56 +45,129 @@ class span_base {
   static_assert(Extent == dynamic_extent, "Only dynamic extent is supported");
 
  public:
-  using element_type    = T;
-  using value_type      = std::remove_cv<T>;
-  using size_type       = std::size_t;
-  using difference_type = std::ptrdiff_t;
-  using pointer         = T*;
-  using iterator        = T*;
-  using const_pointer   = T const*;
-  using reference       = T&;
-  using const_reference = T const&;
-
-  static constexpr std::size_t extent = Extent;
+  using element_type    = T;                  ///< The type of the elements in the span
+  using value_type      = std::remove_cv<T>;  ///< Stored value type
+  using size_type       = std::size_t;        ///< The type used for the size of the span
+  using difference_type = std::ptrdiff_t;     ///< std::ptrdiff_t
+  using pointer         = T*;                 ///< The type of the pointer returned by data()
+  using iterator        = T*;                 ///< The type of the iterator returned by begin()
+  using const_pointer   = T const*;           ///< The type of the pointer returned by data() const
+  using reference       = T&;  ///< The type of the reference returned by operator[](size_type)
+  using const_reference =
+    T const&;  ///< The type of the reference returned by operator[](size_type) const
+
+  static constexpr std::size_t extent = Extent;  ///< The extent of the span
 
   constexpr span_base() noexcept {}
+  /**
+   * @brief Constructs a span from a pointer and a size.
+   *
+   * @param data Pointer to the first element in the span.
+   * @param size The number of elements in the span.
+   */
   constexpr span_base(pointer data, size_type size) : _data(data), _size(size) {}
   // constexpr span_base(pointer begin, pointer end) : _data(begin), _size(end - begin) {}
-  constexpr span_base(span_base const& other) noexcept = default;
-  constexpr span_base& operator=(span_base const& other) noexcept = default;
+  constexpr span_base(span_base const&) noexcept = default;  ///< Copy constructor
+  /**
+   * @brief Copy assignment operator.
+   *
+   * @return Reference to this span.
+   */
+  constexpr span_base& operator=(span_base const&) noexcept = default;
 
   // not noexcept due to undefined behavior when size = 0
+  /**
+   * @brief Returns a reference to the first element in the span.
+   * Calling front on an empty span results in undefined behavior.
+   *
+   * @return Reference to the first element in the span
+   */
   constexpr reference front() const { return _data[0]; }
   // not noexcept due to undefined behavior when size = 0
+  /**
+   * @brief Returns a reference to the last element in the span.
+   * Calling last on an empty span results in undefined behavior.
```

Comment:
```suggestion
   * @brief Returns a reference to the last element in the span.
   *
   * Calling back() on an empty span results in undefined behavior.
```
Missing blank line after `@brief` appears multiple times in this PR.